test/run.py

```diff
@@ -92,6 +92,15 @@ def test_noshell_glob(self):
         self.assertEqual(ec, 0)
         self.assertTrue(all(x in output.lower() for x in ['__init__.py', 'testmodule.py', 'testmodulebis.py']))
 
+    def test_noshell_executable(self):
+        ec, output = run("""%s -c 'print ("foo")'""" % sys.executable)
+        self.assertEqual(ec, 0)
+        self.assertTrue('foo' in output)
+        ec, output = run_simple([sys.executable, '-c', 'print ("foo")'])
```

Comment:
`run_simple` expects a string, while `run` expects and array (you have them the other way around)?

Reply:
no, run_simple also works with an array

Reply:
and run also works with a string, but it's going to be very very escaped